### DISCUSSION TITLE
Stereo anaglyph from side-by-side image in one commad?

### DISCUSSION THREAD

**Discussion by oleglk** [ImageMagick/ImageMagick#6181] [Help] [answered]
### ImageMagick version

6.9.11-60

### Operating system, version and so on

Linux, Windows

### Description

Is it possible to convert a side-by-side stereo image into (red-cyan) anaglyph without explicitly creating intermediate files?

A side-by-side stereo image consists of equal-size left and right images glued together horizontally (left - on the left).

I can do it with two commands like:
####
convert -crop 50%x100%  SBS_IMG.TIF  SBS_IMG.TIF

composite -stereo 0  "SBS_IMG~0.TIF" SBS_IMG~1.TIF"    FCA_IMG.TIF
####

But usually there's a way to make it using a pipe - so that intermediate (left/right) images SBS_IMG~0.TIF and SBS_IMG~1.TIF aren't created. That's what I'm looking for.

Thank you in advance,
Oleg.

**Comment by fmw42** [comment]
Have you tried

`convert stereo_in.suffix -crop 50%x100% +repage +swap -compose -stereo 0 -composite stereo_out.suffix`

If that is not correct, remove the +swap and try again.

**Reply by oleglk** [reply]
Thanks, but so far no luck:
(sorry for the formatting - it gets out of my control)
-----
 convert SBS/DSC03172.jpg  -crop 50%x100% +repage +swap -compose -stereo 0 -composite -quality 90  ANA/DSC03172_FCA_STRAIGHT.JPG

convert-im6.q16: unrecognized compose operator `-stereo' @ error/convert.c/ConvertImageCommand/1067.
------
$ convert SBS/DSC03172.jpg  -crop 50%x100% +repage   -compose -stereo 0 -composite -quality 90  ANA/DSC03172_FCA_STRAIGHT.JPG

convert-im6.q16: unrecognized compose operator `-stereo' @ error/convert.c/ConvertImageCommand/1067.
-------

**Reply by fmw42** [reply]
Probably -stereo only works with composite and not convert ... -compose.   If so, that needs to be fixed so it works either way.

**Comment by fmw42** [comment, marked as answer]
The correct syntax for the convert equivalent of

`composite -stereo 20 lena.jpg zelda1.jpg x.jpg`

![x](https://user-images.githubusercontent.com/13458218/227291726-9ff6707e-0477-4c43-b0ea-23b855477d27.jpg)

is

`convert lena.jpg zelda1.jpg -define compose:args=20 -compose stereo -composite y.jpg`

![y](https://user-images.githubusercontent.com/13458218/227291890-860b66fe-ca02-4c8b-bb12-7b4d16445b98.jpg)

**Reply by oleglk** [reply]
Thanks a lot, it indeed works!

convert SBS/DSC03172.jpg -crop 50%x100% -swap 0,1 -define compose:args=20 -compose stereo -composite -quality 90 ANA/DSC03172_FCA_STRAIGHT.JPG

Note the addition of " -swap 0,1" - otherwise the left and right are inverted!

**Reply by DaVince** [reply]
I'd just like to mention that if your input is JPEG and your output is also JPEG, you had better not do that too often. It'll keep degrading the quality every time you do this. If you have input images that are lossless (like PNG) I would recommend that. ;)
